lib/shared/addon/components/container/form-scheduling/component.js

```diff
@@ -27,96 +19,74 @@ export default Component.extend(ManageLabels, {
   // Is requesting a specific host allowed
   canRequestHost: true,
 
-  // Initial labels and host to start with
-  initialLabels: null,
+  // Initial host to start with
   initialHostId: null,
 
-  // labelArray -> the labels that should be set for the scheduling rules
-
-  // Actions output
-  // setLabels(labelArray)
-  // setRequestedHost(hostId)
-
   // Internal properties
   isRequestedHost: false,
 
   editing: true,
 
-  classNames: ['accordion-wrapper'],
-
-  actions: {
-    addSchedulingRule() {
-      this.send('addAffinityLabel');
-    },
+  advanced: false,
 
-    removeSchedulingRule(obj) {
-      this.send('removeLabel', obj);
-    },
-  },
+  classNames: ['accordion-wrapper'],
 
   init() {
-    this.set('allHosts', this.get('store').all('host'));
     this._super(...arguments);
-
-    this.initLabels(this.get('initialLabels'), 'affinity');
-
-    if ( this.get('isGlobal') )
-    {
-      this.setProperties({
-        isRequestedHost: false,
-        requestedHostId: null,
-      });
-      scheduleOnce('afterRender', () => {
-        this.sendAction('setRequestedHost', null);
-      });
+    this.loadHosts();
+    this.set('advanced', !this.get('editing'));
+    if (this.get('initialHostId')) {
+      debugger
+      this.set('requestedHostId', this.get('initialHostId'));
     }
-    else if ( this.get('initialHostId') )
-    {
-      this.setProperties({
-        isRequestedHost: true,
-        requestedHostId: this.get('initialHostId'),
-      });
+    if (!this.get('scheduling.node')) {
+      this.set('scheduling.node', {});
     }
   },
 
   didReceiveAttrs() {
     if (!this.get('expandFn')) {
-      this.set('expandFn', function(item) {
-          item.toggleProperty('expanded');
+      this.set('expandFn', function (item) {
+        item.toggleProperty('expanded');
       });
     }
   },
 
-  updateLabels(labels) {
-    this.sendAction('setLabels', labels);
+  loadHosts() {
+    const clusterId = this.get('scope.currentCluster.id');
+    this.get('globalStore').findAll('machine', { filter: { clusterId: this.get('scope.currentCluster.id') } }).then((hosts) => {
```

Comment:
Machines are already loaded by one of the parent routes and this is creating a static (non-live) array..  Get a copy of all machines and then add a computed property that depends on it, e.g.

```javascript
_allMachines: null,
init() {
  ...
  set(this, '_allMachines', get(this, 'globalStore').all('machine'));
  ...
}

hostChoices: computed('_allMachines.@each.clusterId', function() {
  return get(this, '_allMachines').filterBy('clusterId', get(this, 'scope.currentCluster.id'));
})